internal/auth0/branding/resource.go

```diff
@@ -147,6 +149,19 @@ func updateBranding(ctx context.Context, d *schema.ResourceData, m interface{})
 		}
 	}
 
+	oldUL, newUL := d.GetChange("universal_login")
+	oldUniversalLogin := oldUL.([]interface{})
+	newUniversalLogin := newUL.([]interface{})
+
+	// This indicates that a removal of the block happened, and we need to delete the template.
+	if len(newUniversalLogin) == 0 && len(oldUniversalLogin) != 0 {
+		if err := api.Branding.DeleteUniversalLogin(); err != nil {
+			return diag.FromErr(err)
+		}
+
+		return readBranding(ctx, d, m)
+	}
```

Comment:
This is the part that actually fixes the GitHub issue around the deletion of the template.